test/test-util.ts

```diff
@@ -31,4 +31,74 @@ test('get should parse the correct tsconfig file', async t => {
   t.deepEqual(contents, FAKE_CONFIG);
 });
 
+
+
+// TODO: test for error due to circular reference
+test('should throw an error if it finds a circular reference', async t => {
+  const FAKE_DIRECTORY = '/some/fake/directory';
+  const FAKE_CONFIG = {files: ['b'], extends: 'CONFIG1'};
+  const CONFIG1 = {extends: 'CONFIG2'};
+  const CONFIG2 = {extends: 'FAKE_CONFIG'};
+  function fakeReadFilep(
+      configPath: string, encoding: string): Promise<string> {
+    switch (configPath) {
+      case 'FAKE_CONFIG':
+
+      case 'CONFIG1':
+        return Promise.resolve(JSON.stringify(CONFIG1));
+      case 'CONFIG2':
+        return Promise.resolve(JSON.stringify(CONFIG2));
+      case '/some/fake/directory/tsconfig.json':
+        return Promise.resolve(JSON.stringify(FAKE_CONFIG));
+      default:
+        return Promise.reject('File Not Found');
+    }
+  }
+  await t.throws(
+      getTSConfig(FAKE_DIRECTORY, fakeReadFilep), Error,
+      'Circular Reference Detected');
+});
+
+
+test('should follow dependency chain caused by extends files', async t => {
+  const FAKE_DIRECTORY = '/some/fake/directory';
+  const FAKE_CONFIG = {
+    compilerOptions: {compileExtraFastPleas: 'SuperSpeed'},
+    files: ['b'],
+    extends: 'CONFIG1'
+  };
+  const CONFIG1 = {include: ['/stuff/*'], extends: 'CONFIG2'};
```

Comment:
Can you add `exclude` and `compilerOptions` to demonstrate that (1) overwriting happens correctly?